share/scan/registry.go

```diff
@@ -219,58 +226,61 @@ func (rc *RegClient) GetImageInfo(ctx context.Context, name, tag string, manifes
 		}
 	}
 
-	// get schema v1
-	manV1, err := rc.Manifest(ctx, name, tag)
-	if err != nil {
-		log.WithFields(log.Fields{"error": err}).Debug("Get Manifest v1 fail")
-	} else {
-		log.WithFields(log.Fields{
-			"mediaType": manV1.SignedManifest.MediaType, "version": manV1.SignedManifest.SchemaVersion, "digest": manV1.Digest,
-			"layers": len(manV1.SignedManifest.FSLayers), "cmds": len(manV1.Cmds), "created": manV1.Created,
-		}).Debug("v1 manifest")
-
-		// Even we send request with accept v1 manifest, we still get v2 format back
-		if manV1.SignedManifest.SchemaVersion <= 1 {
-			if len(manV1.SignedManifest.FSLayers) > 0 {
-				imageInfo.Layers = make([]string, len(manV1.SignedManifest.FSLayers))
-				for i, des := range manV1.SignedManifest.FSLayers {
-					imageInfo.Layers[i] = string(des.BlobSum)
-					// log.WithFields(log.Fields{"i": i, "layer": string(des.BlobSum)}).Debug("v1 manifest")
+	if v2SchemaError != nil {
+		log.WithFields(log.Fields{"name": name, "tag": tag, "error": err.Error()}).Error("could not build v2 image info, trying v1")
```

Comment:
Given that v2SchemaError already has part of the err.Error(), maybe print `v2SchemaError` instead?
```suggestion
		log.WithFields(log.Fields{"name": name, "tag": tag, "error": v2SchemaError.Error()}).Error("could not build v2 image info, trying v1")
```